### PR TITLE
Improve logging and error handling

### LINKED ISSUE #213: Ensure logging output matches Python
We should have a logging message everywhere that the Python agent does. It is okay if we also have additional verbose, helpful output, esp in case of errors.

### PR DESCRIPTION
**Pull request by ansasaki**
Generate log messages equivalent to those generated by the python agent.

Improve error handling and logging for requests.

Fixes #213

### CODE REVIEW

**Comment by lkatalin** [comment]
Looks great so far.

File: src/errors_handler.rs [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+        http::Method::GET => {
+            error = 400;
+            message = format!(
+                "Not Implemented: Use /version or /{}/ interfaces",
+                API_VERSION
+            );
+            response = HttpResponse::BadRequest()
+                .json(JsonWrapper::error(error, &message));
+        }
+        http::Method::POST => {
```
Comment (ashcrow) [inline src/errors_handler.rs:30]: I was going to suggest combining this with the GET match but there is a slight difference in the message.

File: src/errors_handler.rs [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+
+pub(crate) fn path_parser_error(err: PathError, req: &HttpRequest) -> Error {
+    warn!("{} returning 400 response. {}", req.head().method, err);
+
+    HttpResponse::BadRequest()
+        .json(JsonWrapper::error(400, err))
+        .into()
+}
+
+// This handler is ugly as there is no easy way to capture default errors emitted by the server and
```
Comment (ashcrow) [inline src/errors_handler.rs:247]: I appreciate this comment :pray: :+1:

**Review by ashcrow — COMMENTED** [review]
Over all this looks great to me! I'll defer to another reviewer for a second opinion and possible merge.

**Review by lkatalin — APPROVED** [review]
LGTM!

**Pull request merged**